handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -951,6 +965,31 @@ private void wrap(ChannelHandlerContext ctx, boolean inUnwrap) throws SSLExcepti
         }
     }
 
+    static ByteBuf cloneBufferIfReadOnlyAndWantsMutable(ByteBufAllocator allocator, ByteBuf buf,
+                                                        boolean wantsMutableBuffer, boolean wantsDirectBuffer) {
+        if (wantsMutableBuffer && buf != null && buf.isReadOnly()) {
+            return cloneBuffer(allocator, buf, wantsDirectBuffer);
+        }
+        return buf;
+    }
+
+    static ByteBuf cloneBuffer(ByteBufAllocator allocator, ByteBuf buf, boolean wantsDirectBuffer) {
+        ByteBuf copyOfBuf;
```

Comment:
```suggestion
        final ByteBuf copyOfBuf;
```

Reply:
Thanks. Closed this PR as something that isn't needed after creating an integration test in be948fa8c238f77870feea0684893a9415ca0d45
 and investigating more.